dataactbroker/handlers/fileHandler.py

```diff
@@ -1588,5 +1588,37 @@ def fabs_derivations(obj):
     else:
         logging.error("CFDA title not found for CFDA number %s", obj['cfda_number'])
 
+    # deriving awarding agency name
+    awarding_agency_name = sess.query(CGAC).filter_by(cgac_code=obj['awarding_agency_code']).one_or_none()
+    if awarding_agency_name:
+        obj['awarding_agency_name'] = awarding_agency_name.agency_name
+    else:
+        logger.error('Awarding Agency name not found for Awarding Agency Code %s', obj['awarding_agency_code'])
+
+    # deriving awarding sub tier agency name
+    awarding_sub_tier_agency_name = sess.query(SubTierAgency).filter_by(
+        sub_tier_agency_code=obj['awarding_sub_tier_agency_c']).one_or_none()
+    if awarding_sub_tier_agency_name:
+        obj['awarding_sub_tier_agency_n'] = awarding_sub_tier_agency_name.sub_tier_agency_name
+    else:
+        logger.error('Awarding Sub-tier Agency Name not found for Awarding Sub-tier Agency Code %s',
+                     obj['awarding_sub_tier_agency_c'])
+
+    # deriving funding agency name
+    funding_agency_name = sess.query(CGAC).filter_by(cgac_code=obj['funding_agency_code']).one_or_none()
+    if funding_agency_name:
+        obj['funding_agency_name'] = funding_agency_name.agency_name
+    else:
```

Comment:
As with above, we probably don't need this else statement because we do a validity check with rule D20 and the above `if` isn't needed either because of the above suggestion